### DISCUSSION TITLE
[Feature] Facial Recognition: Add to Rando

### DISCUSSION THREAD

**Discussion by bcsteeve** [immich-app/immich#3508] [Feature Request]
### The feature

We need a way to remove faces that just aren't important.  What I've been doing is I created a person named "Rando" and all the random people I don't care about I give that name and merge them together.

A few problems with this: 

Problem No.1) There are SO many randos!  Being limited to merging 5 at a time is extremely slow.
Solution No.1) A "Add unnamed to Rando" button would be awesome, where all the recognized faces on the screen that I have otherwise not assigned a name all get sent to Rando.
Alternative Solution No.1) Never mind Rando, just give us a way to delete/remove (but not "hide") unwanted faces.

Problem No.2) I don't actually want to ever see Rando!  But he's now my MOST important person because there's by far more randos than non-randos!  Hiding him won't work though, because then I can't merge to him.
Solution No.2) Make Rando special.  He should be treated as "hidden" for all purposes except you need to still be able to add to rando quickly and easily.

Potential Problem No.3) Performance?  Potentially, Rando can have tens of thousands of very different faces.  This may (or may not?) cause performance issues if the system is _actually_ trying to find matches. 
Solution No.3) Again, make Rando special.  Once a face is merged with (or otherwise flagged as) a Rando, there's really no need for  they system to try to find matches to him in the future.

When/if there's a way to edit/remove/fix matches, then Rando should be reviewable, I think... but otherwise, I think Rando can be committed to the nether regions of the database and never heard from again.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by EnochPrime** [comment]
What is the benefit of merging all the faces together?

**Reply by bcsteeve** [reply]
Just to get rid of them or make them more manageable.   I haven't seen a better way.

So as I have it right now, I have about 100 "named" faces.  When I click on "All people", it shows those 100 and then tags on another X faces.  It certainly isn't all of them.  If I merge 5 of them into "Rando" another 5 appear.  Maybe my collection is particularly large?  I wouldn't think so, but to give you an idea... Immach has identified over 70,000 faces.  The vast majority of those are just random people in the crowds.

I can go through those and "hide" them, which I did at first but then I end up with 1000's of "hidden" faces littering that page and it makes it impossible to identify new non-Rando faces.  Plus, there are serious performance issues because it still tags on X faces after those.  It becomes unwieldy very quickly.

Merge to rando, delete or whatever... just something to get rid of all these Randos!

**Comment by bcsteeve** [comment]
I took a look at Google Photos to see how it manages this, and it does it both better and worse than Immich.  I'm sure you're aware already how it works.  Until now I honestly never looked at it.

It *only* shows me, my wife, my son and my cats  lol.  It simply ignores the other 70k faces.  But it also ignores my friends and extended family.  It obviously figures out (correctly) who the most important faces are to me.  [update:  I don't really know.  I realize now that I deleted the vast majority of my photos from GP so it may be that I really only have these few faces.  Most of my remaining photos are scenic only]

Then I can click on a photo with a different person and it says:

![image](https://github.com/immich-app/immich/assets/7412219/1016e892-051f-4462-b04c-f59463ca2355)

That's fantastic!


I take it back.  Ignore my feature request and just do that!

**Comment by alextran1502** [comment]
There is an easy way that I think can help you achieve what you want.

When you enter the show/hide face interface, you can choose to hide all the faces and select the important faces that you would like to show. This means your photos will not show a clusters of people that you don't know or care about

**Reply by bcsteeve** [reply]
I get that is the intent, but among those faces ARE faces I want.  Perhaps it is a futile exercise, but I'm trying to go through the 70k faces because there are seemingly hundreds of versions of me, my wife, my son that need to be merged, but they are littered amongst tens of thousands of randos.  

Obviously the system is only showing me a few hundred are a time.  I'll go through those and then more are added.  That's why I started the "Rando" person, so that I could force it to load more when I saw no more that I recognized.

Maybe the algorithm just need to be improved, which I'm sure it will.  Because yet another better solution is for the matches to be more complete.

I notice Google has a training/suggestive-match scheme, which is interesting.  It showed me 6-12 possible matches for each face and asked if they were the same person (they were, 100% of the time)..

I understand that this feature is in its infancy so maybe I'm just expecting too much from it, too soon.

**Reply by alextran1502** [reply]
Fair assessment on the expectation

**Reply by EnochPrime** [reply]
Proposal:
Immich shall set all new faces to hidden until the number of associated images exceeds a threshold.

In this way a majority of the faces will be hidden by default and users can view all to unhide and merge the [hopefully] few cases where an existing face has a different entry.

**Comment by bo0tzz** [comment]
Faces can now be hidden, and there is a setting for a minimum amount of faces detected before a person shows up.
